lib/coach/handler.rb

```diff
@@ -25,16 +25,28 @@ def call(env)
       start = Time.now
 
       publish('coach.handler.start', start_event.dup)
-      response = chain.instrument.call
 
-      finish = Time.now
-      publish('coach.handler.finish',
-              start, finish, nil,
-              start_event.
-                merge(response: { status: response[0] },
-                      metadata: context.fetch(:_metadata, {})))
+      begin
+        response = chain.instrument.call
+      rescue => e
+        finish = Time.now
+        publish('coach.handler.error',
+                start, finish, nil,
+                start_event.merge(
+                  response: { status: 500 },
```

Comment:
My only concern about reusing the same event was that the request didn't truly "finish", i.e. if we're looking at telemetry for an endpoint and it starts to 500 regularly, our logging will show the request has gotten faster overall (e.g. using the average), which is very suspicious.

I'm also kind of unhappy about guessing a 500 - would rather simply remove the response status altogether and fix the downstream places (this might be a major version bump).